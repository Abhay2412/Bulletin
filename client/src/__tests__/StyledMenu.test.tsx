import React from "react";
import { configure, mount, ReactWrapper } from "enzyme";
import Button from "@mui/material/Button";
import StyledMenu from "../components/StyledMenu";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import renderer from "react-test-renderer";
import { store } from "../store";
import { Provider } from "react-redux";

configure({ adapter: new Adapter() });

describe("testing <StyledMenu /> component", () => {
  const wrapper: ReactWrapper = mount(
    <Provider store={store}>
      <StyledMenu />
    </Provider>
  );

  it("should render properly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <StyledMenu />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render exactly one <Button /> component", () => {
    expect(wrapper.find(Button).length).toEqual(1);
  });

  it("should open menu on button click", () => {
    wrapper.find("button").find("#styled-menu-button").simulate("click");
    expect(wrapper.find("#styled-menu-filters-list").exists()).toBeTruthy();
  });
});
