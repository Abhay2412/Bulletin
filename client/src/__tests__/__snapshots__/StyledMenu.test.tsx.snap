// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing <StyledMenu /> component should render properly 1`] = `
<div>
  <button
    aria-haspopup="true"
    className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-zddlty-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    id="styled-menu-button"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "color": "#AA896B",
        "fontSize": "14px",
        "zIndex": 1,
      }
    }
    tabIndex={0}
    type="button"
  >
    Filter
    <span
      className="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="KeyboardArrowDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
        />
      </svg>
    </span>
  </button>
</div>
`;
